docs/rfc/active/5818-serial-updates.md

```diff
@@ -0,0 +1,313 @@
+```json
+{
+  "myList": ["red", "blue", "green", "yellow"]
+}
+```
+
+### Client B
+
+Wants to add "purple" to the list, so the target state is
+```json
+{
+  "myList": ["red", "blue", "green", "purple"]
+}
+```
+
+If they both run their updates in a similar timeframe, both will succeed but either "yellow" or "purple" will be 
+added to the list, not both.
+
+I would like a way for a client to request for an update to be rejected if the initial state differs from its 
+assumptions, i.e. only update the state if the starting state is still the same.
+
+## Motivation
+
+We wish to avoid losing data silently when two clients make updates to the same aspect. This can be quite likely in 
+an event-driven world driving downstream operations on Datasets. We should offer clients the chance to conditionally 
+update an aspect on the basis that what they recently observed is still the case and signal to the client if the 
+state changed before the update was possible. Essentially we need a long "compare-and-swap" operation 
+for atomic updates.
+
+## Requirements
+
+> What specific requirements does your design need to meet? This should ideally be a bulleted list of items you wish
+> to achieve with your design. This can help everyone involved (including yourself!) make sure your design is robust
+> enough to meet these requirements.
+>
+> Once everyone has agreed upon the set of requirements for your design, we can use this list to review the detailed
+> design.
+
+* A client needs to be able to identify the unique state or version of a given aspect when querying for it.
+* The client needs to be able to reference the same state to a GMS "update" endpoint if it wants to ensure the 
+  aspect has not changed between fetching it and mutating it.
+* A client could include multiple aspects in its precondition state, in case one update relies on the state of many 
+  other aspects. We need to defend against race conditions for all the aspects given.
+
+### Extensibility
+
+> Please also call out extensibility requirements. Is this proposal meant to be extended in the future? Are you adding
+> a new API or set of models that others can build on in later? Please list these concerns here as well.
+
+1. The proposal could be extended to include a list of aspect versions which must hold true in order for a single 
+   aspect update to occur.
+2. We could extend to include the state version in a batch update, though this may be complicated if handling 
+   multiple updates to the same aspect in a single batch.
+
+## Non-Requirements
+
+It's not important for us to discuss complex prerequisites here. The only thing we need to enforce is the state of 
+an aspect has not changed when making an update to it.
+
+There are some potential spin-offs of this design which could involve writing some kind of PATCH update where a client 
+supplies a diff instead of a complete new state, but this is out of scope of this particular RFC.
```

Comment:
@RyanHolstien Is currently working on PATCH API support on the GMS ingest path. I'm wondering if we should account for it. I'm assuming that by default PATCH would ignore the version tag